packages/bpk-component-chip-group/src/BpkChipGroup.tsx

```diff
@@ -70,131 +70,172 @@ export type ChipItem = {
   hidden?: boolean;
 } & SingleSelectChipItem;
 
+export type InternalProps = {
+  chipGroupClassNames: string
+};
+
 export type CommonProps = {
   label?: string;
   ariaLabel?: string;
-  type?: ChipGroupType;
   chipStyle?: ChipStyleType;
-  // only required when type = rail
-  leadingNudgerLabel?: string;
-  trailingNudgerLabel?: string;
+  chips: ChipItem[];
+  ariaMultiselectable?: boolean;
 };
 
-export type ChipGroupProps = {
-  chips: ChipItem[];
+export type RailChipGroupProps = {
   stickyChip?: ChipItem;
-  ariaMultiselectable?: boolean;
+  leadingNudgerLabel?: string;
+  trailingNudgerLabel?: string;
+} & CommonProps;
+
+export type WrapChipGroupProps = {
 } & CommonProps;
 
+export type ChipGroupProps = (RailChipGroupProps & { type: typeof CHIP_GROUP_TYPES.rail } | WrapChipGroupProps & { type: typeof CHIP_GROUP_TYPES.wrap });
```

Comment:
We could move the type field into the respective sub-types but it isn't used by any direct consumers so eslint isn't happy with that - defining it with the union here covers what we need without exposing it to places that don't use it

Reply:
Reading this I'm expecting `typeof CHIP_GROUP_TYPES.rail` to be `string`, instead of `'rail'` which is what I think we want?

Reply:
No, thanks to the `as const` on the CHIP_GROUP_TYPES (line 39) the type is specifically `rail` instead of string 

Here is a preview of the evaluated type:
![image](https://github.com/Skyscanner/backpack/assets/3943125/777e4520-f39b-4f17-b60d-9f5d9361f209)